### PR TITLE
Add instructions for creating news fragments; usage and style improvements

### PR DESCRIPTION
**Pull request by ElijahSink**
## Pull Request Check List

- [x] A news fragment is added in `news/` describing what is new.
- [ ] Test cases added for changed code.

## Describe what you have changed in this PR.
Added instructions for creating news fragments and improved overall readability and usage.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,24 +70,30 @@ which is installed in editable mode, or you can use `python -m pdm` from inside
...
-PDM uses `black` coding style and `isort` for sorting import statements, if you are not following them,
-the CI will fail, and your Pull Request will not be merged.
+PDM uses `black` for code style and `isort` for sorting import statements. If you are not following them,
+the CI will fail and your Pull Request will not be merged.
+
+### News fragments
 
+When you make changes such as fixing a bug or adding a feature, you must add a news fragment describing
+your change. News fragments are placed in the `news/` directory, and should be named according to this
+pattern: `<issue_num>.<issue_type>.md` (e.g., `566.bugfix.md`). The contents of the file should be a single sentence in the imperative
```
Comment (frostming) [inline CONTRIBUTING.md:91]: Better to list all available issue types and their meanings, the information can be found at `pyproject.toml` under `tool.towncrier.type` table

File: news/doc.573.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Added instructions to the Contributing section for creating news fragments
```
Comment (frostming) [inline news/doc.573.md:1]: The filename should be `573.doc.md`

Comment (ElijahSink) [reply news/doc.573.md]: Fixed.

File: CONTRIBUTING.md (hunk 1 above)
Comment (ElijahSink) [reply CONTRIBUTING.md]: Done.

**Comment by frostming** [comment]
Thank you! LGTM 🎉

**Pull request merged**